python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -0,0 +1,403 @@
+                    if dataset_config.datalink_url_fmt is None:
+                        raise ValueError(f"dataset type {dataset_type_name} doea not define datalink_url_fmt")
+                    record["access_url"] = dataset_config.datalink_url_fmt.format(**fmt_kws)
+
+                # add extra columns
+                record.update(self.extra_columns)
+
+                batch.add_to_batch(record)
+                if batch.size >= batch_size:
+                    _LOG.debug("Saving next record batch, size=%s", batch.size)
+                    yield batch.make_record_batch()
+
+        # Final batch if anything is there
+        if batch.size > 0:
+            _LOG.debug("Saving final record batch, size=%s", batch.size)
+            yield batch.make_record_batch()
+
+    def _region_to_columns(self, region: Optional[Region], record: Dict[str, Any]) -> None:
+        if region is None:
+            return
+
+        # get spacial parameters from the bounding circle
+        circle = region.getBoundingCircle()
+        center = LonLat(circle.getCenter())
+        record["s_ra"] = center.getLon().asDegrees()
+        record["s_dec"] = center.getLat().asDegrees()
+        record["s_fov"] = circle.getOpeningAngle().asDegrees() * 2
+
+        if isinstance(region, ConvexPolygon):
+            poly = ["POLYGON ICRS"]
+            for vertex in region.getVertices():
+                lon_lat = LonLat(vertex)
+                poly += [
+                    f"{lon_lat.getLon().asDegrees():.6f}",
+                    f"{lon_lat.getLat().asDegrees():.6f}",
+                ]
+            record["s_region"] = " ".join(poly)
+        else:
+            _LOG.warning(f"Unexpected region type: {type(region)}")
+
+    def _exposure_records(self, dimension_record: DimensionRecord, record: Dict[str, Any]) -> None:
+        """Extract all needed info from a visit dimension record."""
+        record["t_exptime"] = dimension_record.exposure_time
+        record["target_name"] = dimension_record.target_name
+
+    def _visit_records(self, dimension_record: DimensionRecord, record: Dict[str, Any]) -> None:
+        """Extract all needed info from an exposure dimension record."""
+        record["t_exptime"] = dimension_record.exposure_time
+        record["target_name"] = dimension_record.target_name
+
+    def _exposure_region(self, dataId: DataCoordinate) -> Optional[Region]:
+        """Return a Region for an exposure.
+
+        This code tries to find a matching visit for an exposure and use the
+        region from that visit.
+        """
+
+        if self._exposure_to_visit is None:
+            self._exposure_to_visit = {}
+            # Read complete relation between visits and exposures. There could
+            # be multiple visits defined per exposure, but they are supposed to
+            # have the same region, so we take one of them at random.
+            records = self.butler.registry.queryDimensionRecords("visit_definition")
```

Comment:
Are we better off doing this query once per `dataId["exposure"]` (constraining the query) and caching for the 189 detectors and other derived dataset types, or doing a potentially huge query that can return millions of records?

Reply:
I think it's more efficient to fetch everything at once if there is a significant chance that you need most of the data from that table (and you have enough memory to keep it all). Doing additional per-exposure or per-visit queries may mean lots of extra round-trips to database, which will likely slow down things significantly. If the size becomes an issue we may need to do something smarter, of course.